src/github.ts

```diff
@@ -0,0 +1,81 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { RestEndpointMethodTypes } from "@octokit/plugin-rest-endpoint-methods";
+
+type ClientType = ReturnType<typeof github.getOctokit>;
+
+export function getClient(token: string): ClientType {
+  return github.getOctokit(token);
+}
+
+export type Output = {
+  title: string;
+  summary: string;
+  annotations: Annotation[];
+};
+
+export type Annotation = {
+  path: string;
+  start_line: number;
+  end_line: number;
+  annotation_level: AnnotationLevel;
+  message: string;
+  title: string;
+  raw_details: string;
+};
+
+export type AnnotationLevel = "notice" | "warning" | "failure";
+
+export async function createCheck(
+  client: ClientType,
+  name: string,
+  annotations: Annotation[]
+): Promise<void> {
+  const output = buildOutput(annotations);
+  const pullRequest = github.context.payload.pull_request;
+  const head_sha = (pullRequest && pullRequest.head.sha) || github.context.sha;
```

Comment:
Could also be written as `pullRequest?.head.sha ?? github.context.sha`. Nullish coalescing operator generally preferred since there is a pitfall using `||` with boolean values.

Reply:
Sure. We're good to use those as a team now?

Reply:
Yeah